### ISSUE TITLE
Document common error codes and their meanings

### ISSUE THREAD

**Issue by ghcs27** [SoCo/SoCo#585]
For some methods (e.g. `pause()`), there are common causes of failure (e.g. nothing to pause) which result in specific error codes (e.g. UPnP Error 701).
We should collect those error codes and document them together with their meaning in the docstring of each method.

Previous conversation about this topic on #454:

@KennethNielsen [commented on 18 Jan 2017](https://github.com/SoCo/SoCo/issues/454#issuecomment-273420182) on #454:
> I can see in the source code that it is documented that e.g. play and pause may raise SoCoUPnPException, but it does not mention anything about the error code. It may be useful to add a little something to the docstrings about the most common sources of failures and about more acceptable sources of failures, such as this one. What do you think.

@ghcs27 [commented on 18 Jan 2017](https://github.com/SoCo/SoCo/issues/454#issuecomment-273514884) on #454:
> I would support adding UPnP error codes and their meaning to the docs. The basic meaning of the error codes can be found on page 89 of the UPnP specs, but some codes probably have a Sonos specific meaning.

@KennethNielsen [commented on 21 Jan 2017[(https://github.com/SoCo/SoCo/issues/454#issuecomment-274274520) on #454:
It might be that we should add some sort of list (probably not exhaustive though, but with links) somewhere in the docs, but my question was more whether it would be a good idea to explain one or 2 particularly relevant error codes in the doc string for methods like e.g. pause. Thoughts?

**Comment by KennethNielsen** [comment]
> We should collect those error codes and document them together with their meaning in the docstring of each method.

I think that is a good suggestion.

One though that we might have is where to document, docstrings vs. read the docs with links from the docstrings. But I tend to think, that it is important that docstrings for the most part, can stand on their own (without online resources) and that the docstring for each method or function, to as large an extent as is reasonable, is an island. If there are lots of repeated information in the docstrings it might be sensible to collect them in a common docstring on the class or module, but never outside the module.

Was this issue meant as a discussion or as a todo item?

**Comment by ghcs27** [comment]
I agree that the docstrings should contain all the information that is relevant and at least somehow specific to the method.
This issue was meant more as a todo item.
